### PR TITLE
[GitHub Release] Polish workflow code

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Description**:
We need to polish the workflow code to make it production ready.

**Changelog**:
- Split the `github-release` stage into `publish-github-release` and `create-github-release`;
- Replace the `checkout` task with passing `--repo` argument to GitHub CLI for the `publicize-github-release` and `check-release-exists` stages;
- Add a description to all workflow inputs;
- Rename `testing` input to `test_run`;
- Format workflow according to [guidelines](https://github.com/dbt-labs/dbt-core/blob/main/.github/_README.md);
- Updated notifications messages to use [workflow commands](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions) so that useful information will pop up on the workflow's annotation block.

**Work results**:
- Build and push draft release: https://github.com/dbt-labs/dbt-core-release-test/actions/runs/3639049102
- Publicize draft release: https://github.com/dbt-labs/dbt-core-release-test/actions/runs/3639107518
- Fail on tag audit: https://github.com/dbt-labs/dbt-core-release-test/actions/runs/3639137398
- Fail on commit audit: https://github.com/dbt-labs/dbt-core-release-test/actions/runs/3639166068

### CODE REVIEW

File: .github/workflows/_github-release.yml [hunk 1]
```diff
@@ -56,7 +59,7 @@ jobs:
           echo The last commit sha in the release: ${{ inputs.sha }}
           echo The release version number: ${{ inputs.version_number }}
           echo Expected Changlog path: ${{ inputs.changelog_path }}
-          echo testing: ${{ inputs.testing }}
+          echo Test run: ${{ inputs.test_run }}
```
Comment (emmyoop) [inline .github/workflows/_github-release.yml:62]: As part of the cleanup, please update these kinds of blocks so the outputs all line up.

For example:
```
echo The last commit sha in the release:  ${{ inputs.sha }}
echo The release version number:          ${{ inputs.version_number }}
echo Expected Changlog path:              ${{ inputs.changelog_path }}
echo Test run:                            ${{ inputs.test_run }}
```

Comment (alexander-smolyakov) [reply .github/workflows/_github-release.yml]: Fixed it.

File: .github/workflows/_github-release.yml [hunk 2]
```diff
@@ -66,149 +75,169 @@ jobs:
...
 
-      - name: Set as draft release
+      - name: "Set As Draft Release"
         id: draft
         run: |
-          if [[ ${{ inputs.testing }} == true ]]
+          if [[ ${{ inputs.test_run }} == true ]]
           then
-            echo This is a draft release
+            echo Release will be publihsed as draft
```
Comment (emmyoop) [inline .github/workflows/_github-release.yml:228]:
```suggestion
            echo Release will be published as draft
```

File: .github/workflows/_github-release.yml [hunk 3]
```diff
@@ -27,16 +27,19 @@ on:
   workflow_call:
     inputs:
```
Comment (emmyoop) [inline .github/workflows/_github-release.yml:28]: Please update the documentation at the top to match these inputs.

File: .github/workflows/_github-release.yml [hunk 4]
```diff
@@ -66,149 +75,169 @@ jobs:
...
+            echo "::error title=${{ env.NOTIFICATION_PREFIX }}: $title::$message"
             exit 1
           fi
           isDraft=$(jq -r '.isDraft' <<< "$output")
-          if [[ $isDraft == true ]] && [[ ${{ inputs.testing }} == false ]]
+          if [[ $isDraft == true ]] && [[ ${{ inputs.test_run }} == false ]]
           then
-            echo Release for tag ${{ steps.set_tag.outputs.tag }} already exists in draft form.  Need to Publish.
+            title="Release tag ${{ steps.set_tag.outputs.tag }} already associated with the draft release."
+            message="Release workflow will publicize the associated release."
```
Comment (emmyoop) [inline .github/workflows/_github-release.yml:114]:
```suggestion
            message="Release workflow will publish the associated release."
```

File: .github/workflows/_github-release.yml [hunk 5]
```diff
@@ -66,149 +75,169 @@ jobs:
...
         run: |
-          echo Tag "${{ steps.check_release_commit.outputs.tag_name}}" already exists for this commit!
-          echo Cannot create a new tag for "${{ needs.check-release-exists.outputs.tag }}" for the same commit.
+          title="Tag ${{ steps.check_release_commit.outputs.tag_name }} already exists for this commit!"
+          message="Cannot create a new tag for ${{ needs.check-release-exists.outputs.tag }} for the same commit"
+          echo "::error title=${{ env.NOTIFICATION_PREFIX }}: $title::$message"
           exit 1
 
-  github-release:
+  publicize-github-release:
```
Comment (emmyoop) [inline .github/workflows/_github-release.yml:179]:
```suggestion
  publish-draft-release:
```

File: .github/workflows/_github-release.yml [hunk 6]
```diff
@@ -66,149 +75,169 @@ jobs:
...
     runs-on: ubuntu-latest
     needs: [check-release-exists, audit-release-different-commit]
+    if: inputs.test_run  == 'false' && needs.check-release-exists.outputs.draft_exists == 'true'
 
     steps:
-      - name: Check out the repository
+      - name: "Publicize Draft Release - ${{ needs.check-release-exists.outputs.tag }}"
```
Comment (emmyoop) [inline .github/workflows/_github-release.yml:185]:
```suggestion
      - name: "Publish Draft Release - ${{ needs.check-release-exists.outputs.tag }}"
```

**Review by emmyoop — CHANGES_REQUESTED** [review]

File: .github/workflows/_github-release.yml (hunk 3 above)
Comment (alexander-smolyakov) [reply .github/workflows/_github-release.yml:29]: Thanks for the point! Fixed in https://github.com/dbt-labs/dbt-release/commit/20a648cd3012d942169b6fca1b0b96159427eb3f.

**Pull request merged**